### ISSUE TITLE
Outdated Ansible version sometimes causes Drush install failure on Windows

### ISSUE THREAD

**Issue by tgroff** [geerlingguy/drupal-vm#47]
Hi Jeff, I've been using drupal-vm on Linux all week with great results. Today is the my first test with drupal-vm on Windows. So far I've not gotten it to work. Here's the error I'm getting:

```
==> drupaldev: TASK: [geerlingguy.drush | Install Drush dependencies with Composer.] ********* 
==> drupaldev: failed: [localhost] => {"failed": true, "rc": 258}                              
==> drupaldev: msg: cannot change to directory '/usr/local/share/drush': path does not exist   
==> drupaldev:                                                                                 
==> drupaldev: FATAL: all hosts have already failed -- aborting                                
==> drupaldev:                                                                                 
==> drupaldev: PLAY RECAP ******************************************************************** 
==> drupaldev:            to retry, use: --limit @/root/playbook.retry                         
==> drupaldev:                                                                                 
==> drupaldev: localhost                  : ok=67   changed=29                                 
==> drupaldev:  unreachable=0    failed=1                                                      
The SSH command responded with a non-zero exit status. Vagrant                                 
assumes that this means the command failed. The output for this command                        
should be in the log above. Please read the output to determine what                           
went wrong.                                                                                    
```

Let me know what I can do to troubleshoot further. I did change to Drupal 7 in `drupal.make.yml` `version: "7.0.x"`.

Windows 7 x64
Git bash shell

Thanks!
--Tony (@drupleg)

**Comment by geerlingguy** [comment]
@tgroff - What version of Ansible are you currently running? (e.g. `ansible --version`). Ansible 1.8.0-1.8.3 had a bug that caused git checkouts to fail in certain cases (like the way the drush checkout is configured). The latest version, 1.8.4, seems to have fixed the problem.

Since Ansible's being installed inside the VM, it might be the repository supplying Ansible in the [JJG-Ansible-Windows](https://github.com/geerlingguy/JJG-Ansible-Windows) project is out of date :-/

**Comment by tgroff** [comment]
Yes, it's Ansible in the VM, v1.8.2

```
==> drupaldev: failed: [localhost] => {"failed": true, "rc": 258}
==> drupaldev: msg: cannot change to directory '/usr/local/share/drush': path does not exist
==> drupaldev:
==> drupaldev: FATAL: all hosts have already failed -- aborting
==> drupaldev:
==> drupaldev: PLAY RECAP ********************************************************************
==> drupaldev:            to retry, use: --limit @/root/playbook.retry
==> drupaldev:
==> drupaldev: localhost                  : ok=64   changed=2    unreachable=0    failed=1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

tgroff@TGROFF-INTEL /v/development/drupal-dev-vm-default (master)
$ vagrant ssh
Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 3.13.0-32-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
Last login: Tue Jan 20 04:15:20 2015 from 10.0.2.2
vagrant@drupaltest:~$ ansible --version
ansible 1.8.2
  configured module search path = None
```

**Comment by geerlingguy** [comment]
Interesting... I though pip had the latest version (the JJG-Ansible-Windows shell script uses pip to install Ansible within the VM), but it seems that's not the case.

Can you add the following line inside `config.yml` and verify it's fixed?

```
drush_keep_updated: yes
```

If so, maybe I'll just leave that line in there for the foreseeable future.

**Comment by tgroff** [comment]
Added `drush_keep_updated: yes` to line 67 of `config.yml`, still fails with the same message.

**Comment by attiks** [comment]
I  ran into the same problem, also with ansible 1.8.2, adding drush_keep_updated worked for me

**Comment by geerlingguy** [comment]
I've put that variable back into the default `example.config.yml`, since it seems on Windows (for some strange reason?), the version of Ansible installed with pip is 1.8.2.

I even tried running the `windows.sh` script on my Mac, inside the VM, on Ubuntu 14.04, and didn't get this issue (it installed Ansible 1.8.4. inside the VM), so I wonder if there's some sort of caching or other issue making Windows users get an older version of Ansible.

Regardless, if you ran into this issue, updating your `config.yml` and adding the line below should hopefully take care of this issue (after `vagrant destroy -f` and `vagrant up` again):

```
drush_keep_updated: yes
```

This is now part of the included `example.config.yml` file, and will be included for a while at least, since I'd rather have an extra default config option than have all Windows users have to bail on the first `vagrant up`!

Thanks @attiks and @tgroff for highlighting this issue!

**Comment by tgroff** [comment]
Hi Jeff, the `drush_keep_updated: yes` doesn't work, I'm still getting the same error and no provisioned environment. 

In troubleshooting further I have found that Ansible isn't getting installed with the drupal-vm vagrant up provision, but ansible-1.8.2 is already installed by default on the box `geerlingguy/ubuntu1404`.
```
# vagrant init geerlingguy/ubuntu1404
# vagrant up
# vagrant ssh
```
will get you just the box with `ansible-1.8.2` already installed. So the drupal-vm playbook isn't in fact installing ansible at all.
```
$ v ssh                                                                          
Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 3.13.0-32-generic x86_64)               
                                                                                 
 * Documentation:  https://help.ubuntu.com/                                      
Last login: Tue Jan 20 04:15:20 2015 from 10.0.2.2
vagrant@vagrant:~$ ansible --version                                             
ansible 1.8.2                                                                    
  configured module search path = None
```

**Comment by geerlingguy** [comment]
Ah, I see! I think I'll have to adjust the `JJG-Ansible-Windows` project to update Ansible if present... I completely forgot I had it pre-installed on the VirtualBox install... and I need to update that box regardless.

**Comment by tgroff** [comment]
Cool, thanks Jeff!

**Comment by geerlingguy** [comment]
I'm working on releasing new versions of the base boxes based on https://github.com/geerlingguy/packer-ubuntu-1404 soon (FYI). Then this issue can be closed.

**Comment by geerlingguy** [comment]
I just published `geerlingguy/ubuntu1404` box version 1.0.3, which includes 14.04.2 LTS, plus Ansible 1.9.1, so this problem shouldn't crop up again. To update your box, cd into the drupal-vm directory and run `vagrant box update`.
